data-raw/R2R_TMH_habitat_inputs/cache_tmh_dataR.R

```diff
@@ -0,0 +1,306 @@
+library(tidyverse)
+
+# file to cache Theoretical Max Habitat data objects 
+# and some exploratory plots to compare SIT existing 
+# to TMH
+
+all_habitat_data <- readRDS('data-raw/R2R_TMH_habitat_inputs/all_habitat_data_for_tmh_inputs_fall_run.rdata')
+
+all_hab_data_long <- all_habitat_data |>
+  mutate(watershed = ifelse(watershed == "Lower San Joaquin River", "San Joaquin River", watershed)) |> 
+  rename(spwn_acres_max = max_spawning_acres,
+         rear_acres_max = max_rearing_acres, 
+         flood_acres_max = max_floodplain_acres) |> 
+  pivot_longer(cols = c(spwn_flow:flood_acres_max), names_to = "metric") |>
+  separate(metric, c('hab', 'unit', 'lifestage'), "_") |>
+  mutate(max_hab = ifelse(lifestage == "max", "max_hab", NA),
+         lifestage = ifelse(lifestage == "max", NA, lifestage),
+         unit = ifelse(unit == "flow", "cfs", unit)) |> 
+  glimpse()
+
+
+# update DSMhabitat values ------------------------------------------------
+
+watersheds <- unique(all_hab_data_long$watershed)
+watersheds <- watersheds[!(watersheds %in%  c('North Delta', "South Delta"))]
+
+## spawning ----------------------------------------------------------------
+
+r_to_r_tmh_fr_spawn <- DSMhabitat::fr_spawn$biop_itp_2018_2019
+
+for(i in 1:length(watersheds)) {
+  ws = watersheds[i]
+  habitat = "spwn"
+  
+  max_hab_acres <- all_hab_data_long |> 
+    filter(watershed == ws & hab == habitat) |> 
+    filter(max_hab == "max_hab") |> 
+    pull(value)
+  
+  existing_acres =   all_hab_data_long |> 
+    filter(watershed == ws & hab == habitat) |> 
+    filter(unit != "cfs") |>  
+    filter(is.na(max_hab)) |> 
+    pull(value)
+  
+  # Note: If the maximum theoretical habitat was less than the existing SIT habitat, 
+  # the theoretical maximum habitat value was used for baseline and model runs. 
```

Comment:
We are not currently using the theoretical max habitat for baseline model runs (when sit hab is higher) - is this what you are stating here?

Reply:
I will fix the the comment - should be "when existing SIT habitat is greater than TMH, we are still using TMH so the WUA would get scaled down"